src/classes/VOL_CTRL_VolunteersCampaignWizard.cls

```diff
@@ -117,267 +119,263 @@ public with sharing class VOL_CTRL_VolunteersCampaignWizard {
     private void CloneExistingJobsAndShifts() {
-        try {
-            list<Volunteer_Job__c> listJobs = new list<Volunteer_Job__c>();
-            list<Volunteer_Shift__c> listShifts = new list<Volunteer_Shift__c>();
-            list<Volunteer_Hours__c> listHours = new list<Volunteer_Hours__c>();
-            list<Volunteer_Job__c> listJobsExisting = new list<Volunteer_Job__c>();
-            list<Volunteer_Shift__c> listShiftsExisting = new list<Volunteer_Shift__c>();
-            list<Volunteer_Hours__c> listHoursExisting = new list<Volunteer_Hours__c>();
-            
-            // initialize the list of all fields on Job.
-            Map<String, Schema.SObjectField> mapS = Schema.SObjectType.Volunteer_Job__c.fields.getMap();
-            list<string> listStrFieldsJob = new list<string>();
-            listStrFieldsJob.addAll(mapS.keySet());       
-            
-            // Ensure the user has access to the object and fields before querying
-            UTIL_Describe.checkReadAccess(UTIL_Describe.StrTokenNSPrefix('Volunteer_Job__c'),
-                new Set<String>(listStrFieldsJob));
-            
-            // find the existing jobs
-            // we need to use dynamic soql to pick up all custom fields we don't know about
-            string strSoql = 'select ';
-            string strComma = '';
+        list<Volunteer_Job__c> listJobs = new list<Volunteer_Job__c>();
+        list<Volunteer_Shift__c> listShifts = new list<Volunteer_Shift__c>();
+        list<Volunteer_Hours__c> listHours = new list<Volunteer_Hours__c>();
+        list<Volunteer_Job__c> listJobsExisting = new list<Volunteer_Job__c>();
+        list<Volunteer_Shift__c> listShiftsExisting = new list<Volunteer_Shift__c>();
+        list<Volunteer_Hours__c> listHoursExisting = new list<Volunteer_Hours__c>();
+
+        // initialize the list of all fields on Job.
+        Map<String, Schema.SObjectField> mapS = Schema.SObjectType.Volunteer_Job__c.fields.getMap();
+        list<string> listStrFieldsJob = new list<string>();
+        listStrFieldsJob.addAll(mapS.keySet());
+
+        // Ensure the user has access to the object and fields before querying
+        UTIL_Describe.checkReadAccess(UTIL_Describe.StrTokenNSPrefix('Volunteer_Job__c'),
+            new Set<String>(listStrFieldsJob));
+
+        // find the existing jobs
+        // we need to use dynamic soql to pick up all custom fields we don't know about
+        string strSoql = 'select ';
+        string strComma = '';
+        for (string strF : listStrFieldsJob) {
+            strSoql += strComma + strF;
+            strComma = ', ';
+        }
+        strSoql += ' from Volunteer_Job__c where Campaign__c = :campaignIdClone order by CreatedDate';
+        listJobsExisting = Database.Query(strSoql);
+
+        // initialize the list of all fields on Shift.
+        mapS = Schema.SObjectType.Volunteer_Shift__c.fields.getMap();
```

Comment:
Since mapS is used for Volunteer_Job__c on line 179, I'd recommend using a different Map for Volunteer_Shift__c.